### PR TITLE
Analytics

### PR DESCRIPTION
**Pull request by Max-Shilin**

### CODE REVIEW

**Comment by dvaletin** [comment]
:star: PR built and available in a preview environment **airpush-repo-single-apk-placement-verificator-pr-6** [here](http://single-apk-placement-verificator.sn-airpush-repo-single-apk-placement-verificator-pr-6.18.236.3.114.nip.io)

**Comment by dvaletin** [comment]
:star: PR built and available in a preview environment **airpush-repo-single-apk-placement-verificator-pr-6** [here](http://single-apk-placement-verificator.sn-airpush-repo-single-apk-placement-verificator-pr-6.18.236.3.114.nip.io)

File: entity/placement.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package entity
+
+type Placement struct {
+	Skipable bool   `json:"skipable"`
+	Type     string `json:"type"`
+	Valid    bool   `json:"valid"`
+	Video    bool   `json:"video"`
+	Airads   AirAds `json:"airads"`
```
Comment (vshibanov) [inline entity/placement.go]: CamelCase --> snake_case  (?)

**Comment by dvaletin** [comment]
:star: PR built and available in a preview environment **airpush-repo-single-apk-placement-verificator-pr-6** [here](http://single-apk-placement-verificator.sn-airpush-repo-single-apk-placement-verificator-pr-6.18.236.3.114.nip.io)

**Comment by dvaletin** [comment]
:star: PR built and available in a preview environment **airpush-repo-single-apk-placement-verificator-pr-6** [here](http://single-apk-placement-verificator.sn-airpush-repo-single-apk-placement-verificator-pr-6.18.236.3.114.nip.io)

File: service/events.go [hunk 2]
```diff
@@ -7,47 +7,61 @@ import (
...
-		return errors.New("placement id is empty")
+func (e Verification) Process(message entity.KafkaPlacementMessage) (*entity.Placement, error) {
+	placement := &entity.Placement{}
+
+	if message.PlacementId == 0 {
+		return placement, errors.New("placement id is empty")
 	}
 
-	stringId := strconv.Itoa(e.PlacementId)
+	stringId := strconv.Itoa(message.PlacementId)
```
Comment (vshibanov) [inline service/events.go]: instead if strconv.Itoa and "+" better to use fmt.Sprintf

File: service/events.go [hunk 3]
```diff
@@ -7,47 +7,61 @@ import (
...
 	})
 	if err != nil {
-		return err
+		return placement, err
 	}
 
 	client := &http.Client{}
-	request, err := http.NewRequest("PUT", apiUrl+stringId, bytes.NewBuffer(jsonBody))
-	request.Header.Set("Authorization", "Bearer "+token)
+	request, err := http.NewRequest("PUT", e.ApiUrl+stringId, bytes.NewBuffer(jsonBody))
```
Comment (vshibanov) [inline service/events.go]: Are you sure what this is analytics opinion to post callback?

File: service/events.go [hunk 4]
```diff
@@ -7,47 +7,61 @@ import (
...
+		return nil, errors.New("Api error response")
+	}
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return placement, err
 	}
 
-	log.Println("Updated for id:" + stringId)
+	log.Println("Updated for placement id:" + stringId)
```
Comment (vshibanov) [inline service/events.go]: may be better to use logrus? (unify tools)

**Comment by dvaletin** [comment]
:star: PR built and available in a preview environment **airpush-repo-single-apk-placement-verificator-pr-6** [here](http://single-apk-placement-verificator.sn-airpush-repo-single-apk-placement-verificator-pr-6.18.236.3.114.nip.io)

**Comment by dvaletin** [comment]
:star: PR built and available in a preview environment **airpush-repo-single-apk-placement-verificator-pr-6** [here](http://single-apk-placement-verificator.sn-airpush-repo-single-apk-placement-verificator-pr-6.18.236.3.114.nip.io)

**Pull request merged**